spec/contract.spec.json

```diff
@@ -0,0 +1,105 @@
+{
+  "id": "truffle-contract-schema.json",
+  "$schema": "http://json-schema.org/schema#",
+  "type": "object",
+  "properties": {
+    "contractName": { "$ref": "#/definitions/ContractName" },
+    "abi": { "$ref": "#/definitions/ABI" },
+    "bytecode": { "allOf": [
+      { "$ref": "#/definitions/Bytecode" },
+      { "description": "Bytecode sent as contract-creation transaction data" }
+    ]},
+    "deployedBytecode": { "allOf": [
+      { "$ref": "#/definitions/Bytecode" },
+      { "description": "On-chain deployed contract bytecode" }
+    ]},
+    "sourceMap": { "allOf": [
+      { "$ref": "#/definitions/SourceMap" },
+      { "description": "Source mapping for contract-creation transaction data bytecode" }
+    ]},
+    "deployedSourceMap": { "allOf": [
+      { "$ref": "#/definitions/SourceMap" },
+      { "description": "Source mapping for contract bytecode" }
+    ]},
+    "source": { "$ref": "#/definitions/Source" },
+    "sourcePath": { "$ref": "#/definitions/SourcePath" },
+    "ast": { "$ref": "#/definitions/AST" },
+    "address": { "$ref": "#/definitions/Address" },
+    "networks": {
+      "patternProperties": {
+        "^[a-zA-Z0-9]+$": { "$ref": "#/definitions/Network" }
+      },
+      "additionalProperties": false
+    },
+    "schemaVersion": { "$ref": "#/definitions/SchemaVersion" },
+    "updatedAt": {
+      "type": "string",
+      "format": "date-time"
+    }
+  },
+  "additionalProperties": false,
+  "required": [
+  ],
+  "definitions": {
+    "ContractName": {
+      "type": "string",
+      "pattern": "^[a-zA-Z_][a-zA-Z0-9_]*$"
+    },
+    "Address": {
+      "type": "string",
+      "pattern": "^0x[a-fA-F0-9]{40}$"
```

Comment:
There's bound to be lots of these _specific-number-of-bytes-in-hex_ regexes... not sure if there's any way around that.